src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs

```diff
@@ -52,7 +53,7 @@ public static partial class JsonSerializer
 
                     if (finishedSerializing)
                     {
-                        if (writer.CurrentDepth == 0)
+                        if (writer.CurrentDepth == 0 || writer.CurrentDepth == currentDepth)
```

Comment:
Lol, I had this fix in my branch :)

https://github.com/dotnet/corefx/commit/c40a3bb6a7e5d5a112c1743d3f4bd6f2229e3362#diff-384f2ed2fc2f34b5d1cc4fa0f93e8cb6R57

Reply:
>Can we remove it?

I tried this and a bunch of tests broke